common/buildcraft/lib/engine/TileEngineBase_BC8.java

```diff
@@ -376,7 +376,7 @@ private void sendPower() {
             return;
         }
         IMjReceiver receiver = getReceiverToPower(tile, currentDirection);
-        if (receiver != null) {
+        if (receiver != null && mjConnector.canConnect(receiver) && receiver.canConnect(mjConnector)) {
```

Comment:
This check is partially implemented in getReceiverToPower -- the addition should go there.